### PR TITLE
use the correct inputmode

### PR DESCRIPTION
**Pull request by bramkragten**
## Type of change

- [ ] Dependency upgrade
- [X] Bugfix (non-breaking change which fixes an issue)
- [ ] New feature (thank you!)
- [ ] Breaking change (fix/feature causing existing functionality to break)
- [ ] Code quality improvements to existing code or addition of tests

## Example configuration

```yaml

```

## Additional information

- This PR fixes or closes issue: fixes #
- This PR is related to issue or discussion:
- Link to documentation pull request:

## Checklist

- [ ] The code change is tested and works locally.
- [ ] There is no commented out code in this PR.
- [ ] Tests have been added to verify that the new code works.

If user exposed functionality or configuration variables are added/changed:

- [ ] Documentation added/updated for [www.home-assistant.io][docs-repository]

[docs-repository]: https://github.com/home-assistant/home-assistant.io

### CODE REVIEW

File: src/components/ha-selector/ha-selector-number.ts [hunk 1]
```diff
@@ -51,8 +51,9 @@ export class HaNumberSelector extends LitElement {
             `
           : ""}
         <ha-textfield
-          inputMode="numeric"
-          pattern="[0-9]+([\\.][0-9]+)?"
+          .inputMode=${(this.selector.number.step || 1) < 1
```
Comment (frenck) [inline src/components/ha-selector/ha-selector-number.ts:54]: A step can be `1.5`, making it a decimal.

Comment (bramkragten) [reply src/components/ha-selector/ha-selector-number.ts:54]: ok, lets just always do decimal then

Comment (frenck) [reply src/components/ha-selector/ha-selector-number.ts:54]: Or:

```suggestion
          .inputMode=${(this.selector.number.step || 1) % 1 != 0
```

Comment (bramkragten) [reply src/components/ha-selector/ha-selector-number.ts:54]: yeah, but 🤷‍♂️ like the only difference is 1 button on the keyboard on mobile 😄

Comment (frenck) [reply src/components/ha-selector/ha-selector-number.ts:54]: Also in validation right? setting integer will not accept floats in the browser?

Comment (bramkragten) [reply src/components/ha-selector/ha-selector-number.ts:54]: No, inputmode doesnt do validation, just for what keyboard is shown on a touchscreen

Comment (bramkragten) [reply src/components/ha-selector/ha-selector-number.ts]: https://developer.mozilla.org/en-US/docs/Web/HTML/Global_attributes/inputmode:

> It's important to understand that the inputmode attribute doesn't cause any validity requirements to be enforced on input. To require that input conforms to a particular data type, choose an appropriate [<input> element type](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/input#input_types). For specific guidance on choosing [<input>](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/input) types, see the [Values](https://developer.mozilla.org/en-US/docs/Web/HTML/Global_attributes/inputmode#values) section.

Comment (frenck) [reply src/components/ha-selector/ha-selector-number.ts]: TIL 👍 Thanks!

**Review by frenck — APPROVED** [review]

**Pull request merged**